src/Umbraco.Core/Persistence/Migrations/Upgrades/TargetVersionSevenSixZero/AddIndexToCmsMemberLoginName.cs

```diff
@@ -4,26 +4,48 @@
 
 namespace Umbraco.Core.Persistence.Migrations.Upgrades.TargetVersionSevenSixZero
 {
-    [Migration("7.6.0", 0, Constants.System.UmbracoMigrationName)]
+    [Migration("7.6.0", 3, Constants.System.UmbracoMigrationName)]
     public class AddIndexToCmsMemberLoginName : MigrationBase
     {
         public AddIndexToCmsMemberLoginName(ISqlSyntaxProvider sqlSyntax, ILogger logger)
             : base(sqlSyntax, logger)
         { }
 
         public override void Up()
-        {
-            var dbIndexes = SqlSyntax.GetDefinedIndexesDefinitions(Context.Database);
-
-            //make sure it doesn't already exist
-            if (dbIndexes.Any(x => x.IndexName.InvariantEquals("IX_cmsMember_LoginName")) == false)
+        {   
+            Execute.Code(database =>
             {
-                Create.Index("IX_cmsMember_LoginName").OnTable("cmsMember")
-                    .OnColumn("LoginName")
-                    .Ascending()
-                    .WithOptions()
-                    .NonClustered();
-            }
+                //Now we need to check if we can actually d6 this because we won't be able to if there's data in there that is too long
+                //http://issues.umbraco.org/issue/U4-9758
+
+                var colLen = (SqlSyntax is MySqlSyntaxProvider)
+                    ? database.ExecuteScalar<int?>("select max(LENGTH(LoginName)) from cmsMember")
+                    : database.ExecuteScalar<int?>("select max(datalength(LoginName)) from cmsMember");
+
+                if (colLen < 900 == false) return null;
```

Comment:
Should we be applying this index no matter what, even if we have no data in the table as of yet?
As I have a test DB where I have no members in it and `colLen` is null as opposed to 0 @Shazwazza ??